src/apps/competitions/models.py

```diff
@@ -511,11 +512,10 @@ def re_run(self, task=None):
         sub.start(tasks=tasks)
         return sub
 
-    def cancel(self):
-        from celery_config import app
+    def cancel(self, status=CANCELLED):
         if self.status not in [Submission.CANCELLED, Submission.FAILED, Submission.FINISHED]:
             app.control.revoke(self.celery_task_id, terminate=True)
-            self.status = Submission.CANCELLED
+            self.status = status
             self.save()
             return True
         return False
```

Comment:
Does this affect the children as well? If the parent submission is set to fail it would make sense to fail and stop all the children.